### ISSUE TITLE
Affinity scoring conflates cross-entropy and MSE for discrete/continuous actions

### ISSUE THREAD

**Issue by sachncs** [sachncs/tsn-affinity#13]
## Problem

`compute_action_affinity_batch` computes cross-entropy for discrete actions and MSE for continuous actions under the same "action affinity" label, and `compute_action_affinity` does the same. Two different losses on the same metric name mean that comparisons across action spaces are meaningless, and downstream selection between models trained on heterogeneous tasks can be biased toward whichever action space produces a numerically smaller loss.

## Evidence

`tsn_affinity/routing/metrics.py:60-70`:

```python
if actions.dim() == 3:
    mse = F.mse_loss(logits, actions, reduction="none").mean(dim=-1)
    mse = (mse * mask).sum() / mask.sum().clamp(min=1.0)
    task_ces[tid].append(float(mse.detach().cpu().item()))
else:
    ce = F.cross_entropy(
        logits.reshape(-1, logits.size(-1)),
        actions.reshape(-1),
        ignore_index=-1,
    )
    task_ces[tid].append(float(ce.detach().cpu().item()))
```

`tsn_affinity/routing/metrics.py:286-297` — same dual-mode behaviour.

Cross-entropy on a 4-class classification task with confident predictions is O(log 1) ≈ 0; MSE on continuous actions is O(sigma^2). Selecting a model copy based on which score is smaller conflates action-space scale with model fit.

## Root Cause

A single function name was overloaded to handle both modalities to avoid duplicating batch-loading logic, but the underlying scoring metric was not disentangled.

## Impact

- `TSNAffinityStrategy` with `affinity_config.mode == "action"` and a mix of discrete and continuous tasks will silently mis-rank copy candidates.
- `tsn_affinity/cli/panda.py` uses continuous actions (Panda), `tsn_affinity/cli/atari.py` uses discrete actions (Atari), and `tsn_affinity/cli/benchmark.py` uses discrete actions (synthetic); a researcher who runs all three and compares affinity numbers from logged results will see inconsistent scales.

## Recommendation

Split into `compute_action_affinity_discrete` (CE) and `compute_action_affinity_continuous` (MSE). The batch loader can be shared, but each public function should return a `ScoreMode` enum-tagged result. Update routing logic in `tsn_affinity/strategies/tsn_affinity.py:_select_copy_for_new_task` to select the appropriate function based on the strategy's declared action type (`self.continuous_actions`).

## Acceptance Criteria

- [ ] Public API exposes two distinct affinity functions
- [ ] `_select_copy_for_new_task` dispatches based on declared action modality
- [ ] Unit test asserts discrete and continuous affinity scores are returned in their own scales (CE ~ O(1), MSE ~ O(action_var))

## Verification

`pytest -k "affinity" -v` should pass; new tests for both modalities.

## Risk

Low. Behaviour-preserving refactor of an internal helper.

**Comment by sachncs** [comment]
Fixed in commit a799277. compute_action_affinity and compute_action_affinity_batch now accept an AffinityMetric argument ('auto', 'cross_entropy', 'mse'). The auto path picks cross-entropy for discrete actions and MSE for continuous actions. Forcing a metric that does not match the action dimensionality raises ValueError. The internal variable is no longer misnamed 'task_ces'.
